File: python/linkedlist/linked-list.py
class Node:
    def __init__(self, data) -> None:
        self.data = data 
        self.next = None

class LinkedList:
    def __init__(self) -> None:
        self.head = None

    def insertAtBegin(self, data):
        new_node = Node(data)
        new_node.next = self.head
        self.head = new_node


    def __repr__(self):
        node = self.head
        nodes = []
        while node is not None:
            nodes.append(node.data)
            node = node.next
        nodes.append("None")
        return " -> ".join(nodes)
    

if __name__ == "__main__" :
    ll = LinkedList()
    ll.insertAtBegin("Buenos Aires")
    ll.insertAtBegin("Quindy")
    ll.insertAtBegin("Berlin")
    print(ll)

File: csharp/LinkedList/LinkedList.cs
class Node
{
    public string data;
    public Node next;

    public Node(string data, Node next = null)
    {
        this.data = data;
        this.next = next;
    }
}

class LinkedList
{
    public Node head;

    public LinkedList(Node head = null)
    {
        this.head = head;
    }

    public void insertAtBegin(string data)
    {
        Node newNode = new Node(data);
        newNode.next = this.head;
        this.head = newNode;
    }

    public override string ToString()
    {
        Node current = this.head;
        List<string> nodes = new List<string>();
        while (current != null)
        {
            nodes.Add(current.data);
            current = current.next;
        }
        nodes.Add("Null");
        return string.Join(" -> ", nodes);
    }
}
